tests/tests/htlc_tests.cpp

```diff
@@ -0,0 +1,408 @@
+   // cler everything out
+   generate_block();
+   trx.clear();
+   // Alice puts a contract on the blockchain
+   {
+      graphene::chain::htlc_create_operation create_operation;
+
+      create_operation.amount = graphene::chain::asset( 10000 );
+      create_operation.destination = bob_id;
+      create_operation.seconds_in_force = 60;
+      create_operation.key_hash = key_hash;
+      create_operation.hash_type = graphene::chain::hash_algorithm::sha256;
+      create_operation.key_size = key_size;
+      create_operation.source = alice_id;
+      trx.operations.push_back(create_operation);
+      sign(trx, alice_private_key);
+      PUSH_TX(db, trx, ~0);
+      trx.clear();
+      graphene::chain::signed_block blk = generate_block();
+      // can we assume that alice's transaction will be the only one in this block?
+      processed_transaction alice_trx = blk.transactions[0];
+      alice_htlc_id = alice_trx.operation_results[0].get<object_id_type>();
+      generate_block();
+   }
+
+   // verify funds on hold (TODO: make sure this can cover fees)
+   BOOST_CHECK_EQUAL( get_balance(alice_id, graphene::chain::asset_id_type()), 90000 );
+
+   // make sure Bob (or anyone) can see the details of the transaction
+   graphene::app::database_api db_api(db);
+   optional<graphene::chain::htlc_object> htlc = db_api.get_htlc(fc::json::to_pretty_string(alice_htlc_id));
+   BOOST_CHECK(htlc);
+
+   // let it expire (wait for timeout)
+   generate_blocks(fc::time_point_sec(120) );
+   // verify funds return (minus the fees)
+   BOOST_CHECK_EQUAL( get_balance(alice_id, graphene::chain::asset_id_type()), 90000 );
+   // verify Bob cannot execute the contract after the fact
+}
+
+BOOST_AUTO_TEST_CASE( htlc_fulfilled )
+{
+   advance_past_hardfork(this);
+	
+   ACTORS((alice)(bob));
+
+   int64_t init_balance(1000000);
+
+   transfer( committee_account, alice_id, graphene::chain::asset(init_balance) );
+   transfer( committee_account, bob_id, graphene::chain::asset(init_balance) );
+
+   uint16_t key_size = 256;
+   std::vector<unsigned char> pre_image(key_size);
+   generate_random_preimage(key_size, pre_image);
+   std::vector<unsigned char> key_hash = hash_it(pre_image);
+
+   graphene::chain::htlc_id_type alice_htlc_id;
+   // cler everything out
+   generate_block();
+   trx.clear();
+   enable_fees();
+   // Alice puts a contract on the blockchain
+   {
+      graphene::chain::htlc_create_operation create_operation;
+
+      create_operation.amount = graphene::chain::asset( 100000 );
+      create_operation.destination = bob_id;
+      create_operation.seconds_in_force = 86400;
+      create_operation.key_hash = key_hash;
+      create_operation.hash_type = graphene::chain::hash_algorithm::sha256;
+      create_operation.key_size = key_size;
+      create_operation.source = alice_id;
+      create_operation.fee = db.current_fee_schedule().calculate_fee( create_operation );
+      trx.operations.push_back(create_operation);
```

Comment:
Please BOOST_CHECK the actual fee on the operation, otherwise it's hard to follow the numbers.